src/RLInterface.jl

```diff
@@ -29,72 +29,86 @@ export
 
 abstract type AbstractEnvironment end
 
-mutable struct MDPEnvironment{S} <: AbstractEnvironment
-    problem::MDP
+"""
+    obsvector_type(::Union{MDP, POMDP})
+
+Returns the type of the observation vector associated with a specific problem. 
+The `MDPEnvironment` and `POMDPEnvironment` wrappers will convert observations to an object of such type when `reset!` or `step!` is called.
+"""
+obsvector_type(::Union{MDP, POMDP}) = Vector{Float32}
```

Comment:
Should we add

```
obsvector_type(::Union{MDP{V,<:Any}, POMDP{<:Any,<:Any,V}}) where V <: AbstractVector = V
```

I am not sure if it's better to have that or `Vector{Float32}`

Reply:
For problems that do not have an `AbstractVector` as a state, would it default to `Vector{Float32}`? 
I think it is better to keep the default simple, I am afraid it might cause some method ambiguity issues as it did with the "convenient" implementations from POMDPModelTools.

Reply:
Yeah, I meant to keep both versions so that problems without `AbstractVector` states/observations would default to `Vector{Float32}`

I think you're probably right - easiest to keep things simple and have everything default to `Vector{Float32}`. Most problems with vector states should work out of the box anyways because `convert(Vector{Float32}, ::V)` is already implemented for them.